dspace-server-webapp/src/test/java/org/dspace/app/rest/LoginAsEPersonIT.java

```diff
@@ -0,0 +1,150 @@
+
+    @Test
+    public void loggedInUserRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                    .param("projection", "full"))
+                    .andExpect(status().isOk())
+                    .andExpect(jsonPath("$._embedded.eperson", EPersonMatcher.matchEPersonOnEmail(admin.getEmail())));
+
+
+    }
+    @Test
+    public void loggedInAsOtherUserRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                    .param("projection", "full")
+                                    .header("X-On-Behalf-Of", eperson.getID()))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$._embedded.eperson",
+                                            EPersonMatcher.matchEPersonOnEmail(eperson.getEmail())));
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserNotAUuidInHeaderBadRequestRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", "not-a-uuid"))
+                        .andExpect(status().isBadRequest());
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserWrongUuidInHeaderBadRequestRetrievalTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", UUID.randomUUID()))
+                        .andExpect(status().isBadRequest());
+
+
+    }
+
+    @Test
+    public void loggedInAsOtherUserNoPermissionForbiddenRetrievalTest() throws Exception {
+
+
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .param("projection", "full")
+                                     .header("X-On-Behalf-Of", eperson.getID()))
+                        .andExpect(status().isForbidden());
+
+
+    }
+
+
+    @Test
+    public void loggedInUserPropertyFalseTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        configurationService.setProperty("webui.user.assumelogin", false);
+        context.restoreAuthSystemState();
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get("/api/authn/status")
+                                     .header("X-On-Behalf-Of", eperson.getID()))
+                        .andExpect(status().isBadRequest());
+
+        context.turnOffAuthorisationSystem();
```

Comment:
I would suggest to don't revert the configuration manually here the change as this is already done in a Before method (just to avoid in future confusion in other developers)